lib/sinon/spy-formatters.js

```diff
@@ -38,6 +39,14 @@ function colorDiffText(diff) {
     return join(objects, "");
 }
 
+function quoteStringValue(value) {
+    if (typeof value === "string" && !value.match(/^'.*'$/)) {
+        return "'" + value + "'";
+    }
```

Comment:
How about simply doing `JSON.stringify(value)` if it's a string? This would handle all escaping edge cases and we don't need the regexp check.

Reply:
That's a straightforward way of doing it, even if it doesn't look as fancy in all cases. I gave it a try, and `"1234"` (literal quote marks included) in this case was stringified as `"\"1234\""`. If that's acceptable, I can go with that.

Reply:
@mroderick Do you have additional thoughts or other ideas?

Reply:
@mantoni Awaiting further comments from @mroderick, does the idea seem good to you?

Reply:
I am working on some experiements to see if I can retire `@sinonjs/formatio` and use Node's `util.inspect` instead. That will do the quoting in the desired way (and save a lot of maintenance).

Reply:
@mroderick If it works, would it make this PR obsolete since the feature would be implemented that way?

Reply:
@rgroothuijsen Yes, that is precisely what it means.